packages/core/src/controllers/relayer.ts

```diff
@@ -305,6 +335,41 @@ export class Relayer extends IRelayer {
   }
 
   // ---------- Private ----------------------------------------------- //
+  /*
+   * In Node, we must detect when the connection is stalled and terminate it.
+   * The logic is, if we don't receive ping from the relay within a certain time, we terminate the connection.
+   * The timer is refreshed on every message received from the relay.
+   *
+   * In the browser, ping/pong events are not exposed, so the above behaviour is handled by `subscribeToNetworkChange` and `isOnline` functions.
+   */
+  private startPingTimeout() {
+    if (!isNode()) return;
+    try {
+      //@ts-expect-error - Types are divergent between the node and browser WS API
+      if (this.provider?.connection?.socket) {
+        //@ts-expect-error
+        this.provider?.connection?.socket?.once("ping", () => {
```

Comment:
```suggestion
        this.provider.connection.socket.once("ping", () => {
```
We don't need the question marks if we're null checking above